src/librustc/hir/def.rs

```diff
@@ -129,6 +129,34 @@ impl DefKind {
             _ => "a",
         }
     }
+
+    pub fn is_valid_in_ns(&self, ns: Namespace) -> bool {
```

Comment:
This might be nicer as a `ns` method that returns the namespace (and the checking can be done by the caller).

Reply:
This only problem is with `Res::Err` which is technically valid in all namespaces. I suppose it could return `Option<Namespace` where None is from `Res::Err`.

Reply:
Seems good as is, it's used only once and the `bool` semantics matches the use case better.

Reply:
`is_valid_in_ns` -> `matches_ns` perhaps?